packages/web-components/@sfx/sayt/src/sayt.ts

```diff
@@ -91,6 +109,36 @@ export default class Sayt extends LitElement {
     if (changedProps.has('visible')) {
       this.hidden = !this.visible;
     }
+    if (changedProps.has('searchbox')) {
+      const oldSearchbox = changedProps.get('searchbox') as string;
+      if (oldSearchbox) {
+        this.toggleSearchboxListener('remove', 'input', this.processSearchboxInput, oldSearchbox);
+      } else {
+        this.toggleSearchboxListener('remove', 'sfx::searchbox_change', this.processSfxSearchboxChange);
+      }
+      if (this.searchbox) {
+        this.toggleSearchboxListener('add', 'input', this.processSearchboxInput, this.searchbox);
+      } else {
+        this.toggleSearchboxListener('add', 'sfx::searchbox_change', this.processSfxSearchboxChange);
+      }
+    }
+  }
+
+  /**
+   * Toggle the events being registered and unregisterd when the `searchbox` property changes.
+   *
+   * @param toggle A string to indicate the type of eventListener(add or remove).
+   * @param eventName The name of the eventListener to register or unregister.
+   * @param eventCallback The name of the callback function triggered by the eventListener.
+   * @param searchboxType The `oldSearchbox` or `this.searchbox`.
+   */
+  toggleSearchboxListener(toggle: string, eventName: string, eventCallback: any, searchboxType?: string) :void {
+    if (searchboxType) {
+      const searchbox :any = document.getElementById(searchboxType);
+      if (searchbox) searchbox[`${toggle}EventListener`](eventName, eventCallback);
+    } else {
+      (window as any)[`${toggle}EventListener`](eventName, eventCallback);
```

Comment:
I don't think `window` needs to be cast here.

```suggestion
      window[`${toggle}EventListener`](eventName, eventCallback);
```